go-controller/pkg/cluster/gateway_localnet.go

```diff
@@ -92,15 +100,38 @@ func generateGatewayNATRules(ifname string, clusterCIDRs []string) []iptRule {
 		args:  []string{"-i", ifname, "-m", "comment", "--comment", "from OVN to localhost", "-j", "ACCEPT"},
 	})
 
+	rules = append(rules, iptRule{
+		table: "mangle",
+		chain: "PREROUTING",
+		args:  []string{"-j", "CONNMARK", "--restore-mark"},
+	})
+	rules = append(rules, iptRule{
+		table: "mangle",
+		chain: "PREROUTING",
+		args:  []string{"-m", "mark", "!", "--mark", "0x0", "-j", "ACCEPT"},
+	})
+
 	for _, cidr := range clusterCIDRs {
-		// NAT for the interface
 		rules = append(rules, iptRule{
-			table: "nat",
-			chain: "POSTROUTING",
-			args:  []string{"-s", cidr, "-j", "MASQUERADE"},
+			table: "mangle",
+			chain: "PREROUTING",
+			args:  []string{"-s", cidr, "-i", "br-nexthop", "-m", "mark", "--mark", "0x0", "-j", "MARK", "--set-xmark", natFwMark},
 		})
 	}
 
+	rules = append(rules, iptRule{
+		table: "mangle",
+		chain: "PREROUTING",
+		args:  []string{"-j", "CONNMARK", "--save-mark"},
+	})
```

Comment:
can we add -- "-i", "br-nexthop" -- to the rule at L125, so that the rule only applies on that interface.